### DISCUSSION TITLE
Make Owner:aimpos() ignore entities just like ranger also some other things

### DISCUSSION THREAD

**Discussion by DecVes** [wiremod/wire#3001] [Suggestions]
I love aimpos and I use it for turrets and ballistics and all sorts of things but when I try to make a vehicle gun It looks at the prop your aimpos is on which is fine but most of the time is it not, What i'm asking is that you can use Adv entity marker to create an array of objects and the aimpos will go right through them, Now this same principle would be nice on Cam controller as well and some other things that act like aimpos and camhitpos.

now dont make it so the prop turns invisible or anything like that just make it so aimpos goes through it instead and the player can put materials or whatever on it

![ENT](https://github.com/wiremod/wire/assets/115730620/9ae47ca0-bb83-4da4-9b7c-fa7e19052c2c)

**Comment by DecVes** [comment]
does anyone see this

**Reply by vurvdev** [reply]
Yes, I saw your post but I don't really get what you're saying with the description and the visual, sorry

For the title part, if you just want a custom filtered :aimPos() / :eyeTrace(), then make your own rangerOffset at `owner():shootPos()` toward `owner():eye()`. Gmod has no way to add filters to the aimPos/eyeTrace trace

**Reply by DecVes** [reply]
yeah sorry its kinda hard to read but yeah I want it so the Aimpos/eyetrace can go through certain props so the aimpos doesnt mess some things up, now for the ranger offset I could probably make one of those for now.

**Comment by Divran** [comment]
If you want your trace to be filtered you will need to use E2's rangerOffset functions that accept filtering

Reason why is:

`Ply:aimPos()` is just a shorthand for `Ply:eyeTrace():pos()`, see <https://github.com/wiremod/wire/blob/master/lua/entities/gmod_wire_expression2/core/player.lua#L688-L693> and <https://github.com/wiremod/wire/blob/master/lua/entities/gmod_wire_expression2/core/ranger.lua#L337-L343>

and `GetEyeTraceNoCursor` is a function provided by gmod, see <https://wiki.facepunch.com/gmod/Player:GetEyeTraceNoCursor>
Gmod itself always runs a trace once per tick and `GetEyeTrace` simply allows you to retrieve the results that gmod of that trace. This saves a ton of performance because it means all 200 or whatever addons you install don't have to run their own individual traces every time they need it, they can just get one that was already done anyway.
